### DISCUSSION TITLE
[5차 스프린트] 데모데이 준비하며 발견한 버그들과 앞으로 해야할 것들!

### DISCUSSION THREAD

**Discussion by Minjoo522** [woowacourse-teams/2024-devel-up#579] [백로그 논의]
#### 논의가 필요한 내용

데모데이 준비하며 발견한 버그들과 앞으로 해야할 것들!

**Comment by Minjoo522** [comment]
- 디스커션 조회 FETCH JOIN 문제 해결 -> 연관 관계 끊기
- 개발 서버도 Read / Write DB 분리 해야하나?
- Read / Write 분리 설정 파일 사소한 리팩토링
- 페이지네이션
- 디스커션 : 미션 없을 때, 해시태그 없을 때 테스트 코드 추가
- 신기능 : 인기 디스커션, 인기 미션, 인기 솔루션

---
### 코치님 피드백

- 배너가 너무 크다. && 캐러샐에 대체 뭐가 들어갈거냐?
- 미션의 개수가 많아지면 해시태그를 어떻게 보여줄 것인가?
- 디스커션 작성할 때 쿼리가 세 번 나가는 것인지?
- 대댓글 카운트 안하는지?
- 🚨 제출한 디스커션에 내 디스커션 다 조회 안되는 버그 있음 -> 딱 하나만 조회됨 ..ㄷㄷ
   
![image](https://github.com/user-attachments/assets/d7b41427-9323-40d8-be13-f9e603dbc3f6)

- 오류 페이지 디자인 통일시키기
- 고민한 부분들을 조금 더 공유하기
- 실제 상황처럼 데이터를 고르지 않고 편향된 데이터를 넣어봐야 한다. -> 특정 미션, 특정 해시태그
- 데이터 조회 시 목표 수치를 정해보기
- 이상하면 지레짐작해서 하지 말고 물어보자!
- 처음 진입점이 모바일일 가능성이 높다-> 기능 동작이 다 되지는 않아도 모바일 대응을 조금이라도 해야한다. 아니면 모바일에서 지원되지 않는다고 메시지를 띄우자 -> 소개 페이지만이라도 보이고 PC로 이용 유도
- 프론트 성능 개선 : before -> after 쓰는 입장에서 어떻게 좋아졌는지? -> 쓰는 입장에서 문제가 없다고 판단되면 유지해도 상관 없으니까 시간이 드는 작업이기 때문에 유저 입장에서 명확한 아웃풋으로 설득을해야. 우리 서비스에 도움이 됐다는 대전제가 있어야한다. 왜?했는지가 중요하다. + 목표 잡을 때 백엔드랑 같이 합의해서 최적화 -> 프론트 + 백 합쳐져서 개선이 이루어져야 한다. -> 실제 서비스에 도움이 되는 작업이 될 수 있도록

---
### 6차 스프린트

- [ ] 다른 팀 크루들 몇 명 붙잡고 써보게 하고 인터뷰 해봐야한다.

**Comment by chosim-dvlpr** [comment]
### 피드백 정리 (프론트 위주)

디스커션 제출 후 대시보드에 내가 작성한 디스커션에 업데이트가 안됨

캐러셀 -> 배너로 (현재 컨텐츠는 하나 뿐인데 캐러셀보다는 배너가 좋을 것 같다)
- 처음 들어오는 사용자를 위해 튜토리얼을 화면에 띄우는 것도 좋을 듯

처음 진입점은 모바일일 경우가 많음 (모바일로 공유되기 때문에)
- 모바일에서는 실행x 등 서비스 안된다고 보여주기 - 서비스는 깨지지 않아야 한다
- 소개 페이지 정도는 보여주면 좋겠다 => PC로 이용하도록 유도

번들 사이즈 줄이기 관련 피드백
- 기준점을 수치가 아닌 시간으로 재면 좋음 - 사용자 입장에서, 번들 사이즈를 줄였을 때 명확한 결과물, 서비스의 어떤 점에 도움이 되었는지, 어떤 목표를 달성했는지
- 사용자 시나리오를 정해서 성능 개선을 해보는 것도 좋음 

요청 수 30개 이하 관련 피드백
- HTTP 1 ? 막혀있는지 - 백엔드와 함께 논의
- ~페이지에서 ~한 작업을 실행했을 때 ~요청 수가 나오고 로드되는데 ~초가 걸린다

팀끼리 팀 프로젝트에 쏟는 시간을 정하는 규칙도 만들면 좋음
